cmd/sourced/cmd/compose.go

```diff
@@ -66,16 +68,38 @@ type composeSetDefaultCmd struct {
 	Command `name:"set" short-description:"Set the active docker compose file" long-description:"Set the active docker compose file"`
 
 	Args struct {
-		Version string `positional-arg-name:"version" description:"Either a revision (tag, full sha1) or a URL to a docker-compose.yml file"`
+		File string `positional-arg-name:"index/name" description:"Provide name or index of compose file on 'sourced compose list'"`
 	} `positional-args:"yes" required:"yes"`
 }
 
 func (c *composeSetDefaultCmd) Execute(args []string) error {
-	err := composefile.SetActive(c.Args.Version)
+	files, err := composefile.List()
+
 	if err != nil {
 		return err
 	}
 
+	index, err := strconv.Atoi(c.Args.File)
+
+	if err == nil {
+		if index >= 0 && index < len(files) {
+			active := files[index]
+			err = composefile.SetActive(active)
+		} else {
+			fmt.Println("Index is out of range of the files in 'sourced compose list'")
+			return nil
+		}
+		if err != nil {
+			return err
+		}
+
+	} else if err != nil {
```

Comment:
`if err != nil` is not necessary.

Reply:
It was a fast fix, so i missed this state. I am gonna fix it. Thanks.